libs/datasets/combined_datasets.py

```diff
@@ -47,17 +53,82 @@ class RegionLatestNotFound(IndexError):
     pass
 
 
+RegionMaskOrRegions = NewType("RegionMaskOrRegions", Union[RegionMask, Region, Collection[Region]])
+
+
+@final
+@dataclass(frozen=True)
+class DataSourceAndRegionMasks:
+    """Represents a DataSource class and include/exclude region masks.
+
+    Use function `datasource_regions` to create instance of this class.
+
+    Instances of this class can be used in the same places where the DataSource type (not
+    instances of the DataSource) are used.
+
+    Using this class depends on an existing source of all location_ids. Currently it depends on
+    the existing combined data and is used to produce a new combined data. A recursive data
+    dependency isn't great but works and we don't otherwise have a stable source of locations to
+    use for filtering. Fixing this could be part of
+    https://trello.com/c/2Pa4DMyu/836-trim-down-the-timeseriesindexfields
+    """
+
+    data_source_cls: Type[data_source.DataSource]
+    include: Optional[RegionMaskOrRegions] = None
+    exclude: Optional[RegionMaskOrRegions] = None
+
+    @property
+    def EXPECTED_FIELDS(self):
+        """Returns the same property of the wrapped DataSource class."""
+        return self.data_source_cls.EXPECTED_FIELDS
+
+    @property
+    def SOURCE_NAME(self):
```

Comment:
Generally the convention is that only class variables are caps.  Would you be opposed to be lowercase?

Reply:
Ahh I missed this - yeah I think that's alright for this case then - maybe make a comment saying that? This is pretty much implementing an interface